images/linux/scripts/installers/azpowershell.sh

```diff
@@ -8,90 +8,36 @@
-if ! pwsh -Command '$actualPSModulePath = $env:PSModulePath ; $env:PSModulePath = "/usr/share/az_1.0.0:" + $env:PSModulePath;
-    if (!(get-module -listavailable -name Az.accounts)) {
-        Write-Host "Az Module was not installed"; $env:PSModulePath = $actualPSModulePath; exit 1
-    }
-    $env:PSModulePath = $actualPSModulePath
-    $actualPSModulePath = $env:PSModulePath ; $env:PSModulePath = "/usr/share/az_1.6.0:" + $env:PSModulePath;
-    if (!(get-module -listavailable -name Az.accounts)) {
-        Write-Host "Az Module was not installed"; $env:PSModulePath = $actualPSModulePath; exit 1
-    }
-    $env:PSModulePath = $actualPSModulePath
-    $actualPSModulePath = $env:PSModulePath ; $env:PSModulePath = "/usr/share/az_2.3.2:" + $env:PSModulePath;
-    if (!(get-module -listavailable -name Az.accounts)) {
-        Write-Host "Az Module was not installed"; $env:PSModulePath = $actualPSModulePath; exit 1
-    }
-    $env:PSModulePath = $actualPSModulePath
-    $actualPSModulePath = $env:PSModulePath ; $env:PSModulePath = "/usr/share/az_2.6.0:" + $env:PSModulePath;
-    if (!(get-module -listavailable -name Az.accounts)) {
-        Write-Host "Az Module was not installed"; $env:PSModulePath = $actualPSModulePath; exit 1
-    }
-    $env:PSModulePath = $actualPSModulePath
-    $actualPSModulePath = $env:PSModulePath ; $env:PSModulePath = "/usr/share/az_2.8.0:" + $env:PSModulePath;
-    if (!(get-module -listavailable -name Az.accounts)) {
-        Write-Host "Az Module was not installed"; $env:PSModulePath = $actualPSModulePath; exit 1
-    }
-    $env:PSModulePath = $actualPSModulePath
-    $actualPSModulePath = $env:PSModulePath ; $env:PSModulePath = "/usr/share/az_3.1.0:" + $env:PSModulePath;
-    if (!(get-module -listavailable -name Az.accounts)) {
-        Write-Host "Az Module was not installed"; $env:PSModulePath = $actualPSModulePath; exit 1
-    }
-    $env:PSModulePath = $actualPSModulePath
-    $actualPSModulePath = $env:PSModulePath ; $env:PSModulePath = "/usr/share/az_3.5.0:" + $env:PSModulePath;
-    if (!(get-module -listavailable -name Az.accounts)) {
-        Write-Host "Az Module was not installed"; $env:PSModulePath = $actualPSModulePath; exit 1
-    }
-    $env:PSModulePath = $actualPSModulePath
-    $actualPSModulePath = $env:PSModulePath ; $env:PSModulePath = "/usr/share/az_3.8.0:" + $env:PSModulePath;
-    if (!(get-module -listavailable -name Az.accounts)) {
-        Write-Host "Az Module was not installed"; $env:PSModulePath = $actualPSModulePath; exit 1
-    }
-    $env:PSModulePath = $actualPSModulePath'; then
-    exit 1
-fi
+for version in ${versions[@]}; do
+    modulePath="/usr/share/az_$version"
+    pwsh -Command "
+        \$env:PSModulePath = '${modulePath}:' + \$env:PSModulePath;
+        if ( -not (Get-Module -ListAvailable -Name Az.Accounts)) {
+            Write-Host 'Az Module was not installed'
+            exit 1
+        }"
+    if [ $? -ne 0 ]; then

```

Comment:
exit 1 from powershell doesn't break the build?

Reply:
Good point, but `set -e` is not commonly used, so let's leave IF condition as it is.